src/Codeception/Lib/GroupManager.php

```diff
@@ -118,6 +118,12 @@ public function groupsForTest(\PHPUnit\Framework\Test $test)
                 if (strpos($filename . ':' . $test->getName(false), $testPattern) === 0) {
                     $groups[] = $group;
                 }
+                // TODO: consider mb_strtolower per https://stackoverflow.com/a/5473569
+                if (method_exists($test, 'getMetadata')
+                    && strcasecmp($filename . ':' . $test->getMetadata()->getFeature(), $testPattern) === 0
```

Comment:
This needs to be UFT-8 compliant as put in your TODO
`strcasecmp($filename . ':' . $test->getMetadata()->getFeature(), $testPattern) === 0`to be replaced by `mb_strtolower($filename . ':' . $test->getMetadata()->getFeature()) === mb_strtolower($testPattern)`